protocol/testing/e2e/gov/perpetuals_test.go

```diff
@@ -0,0 +1,305 @@
+package gov_test
+
+import (
+	"testing"
+
+	"github.com/cometbft/cometbft/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	"github.com/stretchr/testify/require"
+)
+
+// This tests `MsgUpdateParams` in `x/perpetuals`.
+func TestUpdatePerpetualsModuleParams(t *testing.T) {
+	tests := map[string]struct {
+		msg                      *perptypes.MsgUpdateParams
+		expectedProposalStatus   govtypesv1.ProposalStatus
+		expectSubmitProposalFail bool
+	}{
+		"Success": {
+			msg: &perptypes.MsgUpdateParams{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Params: perptypes.Params{
+					FundingRateClampFactorPpm: 123_456,
+					PremiumVoteClampFactorPpm: 123_456_789,
+					MinNumVotesPerSample:      15,
+				},
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Fail: invalid authority": {
```

Comment:
In general, could we add some other failure tests other than `invalid authority`?